modules/gateway/gateway_test.go

```diff
@@ -17,11 +17,6 @@ func newTestingGateway(name string, t *testing.T) *Gateway {
 	if err != nil {
 		t.Fatal(err)
 	}
-	// Manually add myAddr as a node. This is necessary because g.addNode
-	// rejects loopback addresses.
-	id := g.mu.Lock()
-	g.nodes[g.myAddr] = struct{}{}
-	g.mu.Unlock(id)
```

Comment:
Good idea. Although in this specific case I think the assertion would be slightly different. It's not that we are assuming `g.nodes` contains `g.myAddr` (it shouldn't, this was just a kludge), but rather we were assuming `addNode` would reject loopback addresses.
```go
if err := g.addNode("localhost:123"); err == nil {
    t.Fatal("assumption invalidated - addNode accepted a loopback address:", err)
}
```